.github/workflows/shrinkwrap.yml

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2020 - 2024 Pionix GmbH and Contributors to EVerest
+
+
+name: Generate and Publish npm-shrinkwrap
+# The purpose of this job is to generate an npm-shrinkwrap.json which is needed for the yocto layer
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  create-shrinkwrap:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Generate npm-shrinkwrap.json
+        run: |
+          npm install
```

Comment:
Don't we want to use `npm ci` here?

Reply:
The problem is, that npm ci depends on a package-lock.json which we don't have, as we are using npm. We use npm install to generate the package-lock.json and after that we can use the npm shrinkwrap option.